### PR TITLE
remove g_mediaManager global

### PR DESCRIPTION
**Pull request by lrusak**
A trivial change. This moves the CMediaManager initialization into CServiceManager.

a small part of my larger iso rework.

### CODE REVIEW

File: xbmc/cores/VideoPlayer/DVDInputStreams/DVDFactoryInputStream.cpp [hunk 1]
```diff
@@ -75,7 +75,7 @@ std::shared_ptr<CDVDInputStream> CDVDFactoryInputStream::CreateInputStream(IVide
   }
 
 #ifdef HAS_DVD_DRIVE
-  if(file.compare(g_mediaManager.TranslateDevicePath("")) == 0)
+  if(file.compare(CServiceBroker::GetMediaManager().TranslateDevicePath("")) == 0)
```
Comment (Rechi) [inline xbmc/cores/VideoPlayer/DVDInputStreams/DVDFactoryInputStream.cpp]: scape missing between `if` and `(`

File: xbmc/dialogs/GUIDialogFileBrowser.cpp [hunk 2]
```diff
@@ -557,7 +557,7 @@ bool CGUIDialogFileBrowser::HaveDiscOrConnection( int iDriveType )
 {
   if ( iDriveType == CMediaSource::SOURCE_TYPE_DVD )
   {
-    if ( !g_mediaManager.IsDiscInDrive() )
+    if ( !CServiceBroker::GetMediaManager().IsDiscInDrive() )
```
Comment (Rechi) [inline xbmc/dialogs/GUIDialogFileBrowser.cpp]: no space after `(` and before `)`

File: xbmc/dialogs/GUIDialogFileBrowser.cpp [hunk 3]
```diff
@@ -936,7 +936,7 @@ bool CGUIDialogFileBrowser::OnPopupMenu(int iItem)
       VECSOURCES shares=m_shares;
       if (CGUIDialogNetworkSetup::ShowAndGetNetworkAddress(newPath))
       {
-        g_mediaManager.SetLocationPath(strOldPath,newPath);
+        CServiceBroker::GetMediaManager().SetLocationPath(strOldPath,newPath);
```
Comment (Rechi) [inline xbmc/dialogs/GUIDialogFileBrowser.cpp]: space is missing after `,`

File: xbmc/filesystem/VirtualDirectory.cpp [hunk 4]
```diff
@@ -175,19 +176,19 @@ void CVirtualDirectory::GetSources(VECSOURCES &shares) const
     CMediaSource& share = shares[i];
     if (share.m_iDriveType == CMediaSource::SOURCE_TYPE_DVD)
     {
-      if(g_mediaManager.IsAudio(share.strPath))
+      if(CServiceBroker::GetMediaManager().IsAudio(share.strPath))
```
Comment (Rechi) [inline xbmc/filesystem/VirtualDirectory.cpp]: space is missing between `if` and `(`

File: xbmc/guilib/guiinfo/SystemGUIInfo.cpp [hunk 5]
```diff
@@ -482,13 +482,13 @@ bool CSystemGUIInfo::GetBool(bool& value, const CGUIListItem *gitem, int context
...
-      value = g_mediaManager.IsDiscInDrive();
+      value = CServiceBroker::GetMediaManager().IsDiscInDrive();
       return true;
     case SYSTEM_MEDIA_AUDIO_CD:
     #ifdef HAS_DVD_DRIVE
-      if (g_mediaManager.IsDiscInDrive())
+      if (CServiceBroker::GetMediaManager().IsDiscInDrive())
       {
-        MEDIA_DETECT::CCdInfo *pCdInfo = g_mediaManager.GetCdInfo();
+        MEDIA_DETECT::CCdInfo *pCdInfo = CServiceBroker::GetMediaManager().GetCdInfo();
```
Comment (Rechi) [inline xbmc/guilib/guiinfo/SystemGUIInfo.cpp]: left align `*`

File: xbmc/music/windows/GUIWindowMusicBase.cpp [hunk 6]
```diff
@@ -568,18 +568,18 @@ void CGUIWindowMusicBase::GetContextButtons(int itemNumber, CContextButtons &but
       }
 #ifdef HAS_DVD_DRIVE
       // enable Rip CD Audio or Track button if we have an audio disc
-      if (g_mediaManager.IsDiscInDrive() && m_vecItems->IsCDDA())
+      if (CServiceBroker::GetMediaManager().IsDiscInDrive() && m_vecItems->IsCDDA())
       {
         // those cds can also include Audio Tracks: CDExtra and MixedMode!
-        MEDIA_DETECT::CCdInfo *pCdInfo = g_mediaManager.GetCdInfo();
+        MEDIA_DETECT::CCdInfo *pCdInfo = CServiceBroker::GetMediaManager().GetCdInfo();
```
Comment (Rechi) [inline xbmc/music/windows/GUIWindowMusicBase.cpp]: left align `*`

File: xbmc/music/windows/GUIWindowMusicBase.cpp [hunk 7]
```diff
@@ -699,7 +699,7 @@ bool CGUIWindowMusicBase::OnAddMediaSource()
 
 void CGUIWindowMusicBase::OnRipCD()
 {
-  if(g_mediaManager.IsAudio())
+  if(CServiceBroker::GetMediaManager().IsAudio())
```
Comment (Rechi) [inline xbmc/music/windows/GUIWindowMusicBase.cpp]: space is missing between `if` and `(`

File: xbmc/music/windows/GUIWindowMusicBase.cpp [hunk 8]
```diff
@@ -714,7 +714,7 @@ void CGUIWindowMusicBase::OnRipCD()
 
 void CGUIWindowMusicBase::OnRipTrack(int iItem)
 {
-  if(g_mediaManager.IsAudio())
+  if(CServiceBroker::GetMediaManager().IsAudio())
```
Comment (Rechi) [inline xbmc/music/windows/GUIWindowMusicBase.cpp]: space is missing between `if` and `(`

File: xbmc/music/windows/GUIWindowMusicNav.cpp [hunk 9]
```diff
@@ -586,10 +586,10 @@ void CGUIWindowMusicNav::GetContextButtons(int itemNumber, CContextButtons &butt
       CGUIDialogContextMenu::GetContextButtons("music", item, buttons);
 #ifdef HAS_DVD_DRIVE
       // enable Rip CD an audio disc
-      if (g_mediaManager.IsDiscInDrive() && item->IsCDDA())
+      if (CServiceBroker::GetMediaManager().IsDiscInDrive() && item->IsCDDA())
       {
         // those cds can also include Audio Tracks: CDExtra and MixedMode!
-        MEDIA_DETECT::CCdInfo *pCdInfo = g_mediaManager.GetCdInfo();
+        MEDIA_DETECT::CCdInfo *pCdInfo = CServiceBroker::GetMediaManager().GetCdInfo();
```
Comment (Rechi) [inline xbmc/music/windows/GUIWindowMusicNav.cpp]: left align `*`

File: xbmc/platform/win32/storage/Win32StorageProvider.cpp [hunk 10]
```diff
@@ -31,15 +32,15 @@ void CWin32StorageProvider::Initialize()
...
+    CServiceBroker::GetMediaManager().SetHasOpticalDrive(true);
   else
     CLog::Log(LOGDEBUG, "%s: No optical drive found.", __FUNCTION__);
 
 #ifdef HAS_DVD_DRIVE
   // Can be removed once the StorageHandler supports optical media
   VECSOURCES::const_iterator it;
   for(it=vShare.begin();it!=vShare.end();++it)
-    if(g_mediaManager.GetDriveStatus(it->strPath) == DRIVE_CLOSED_MEDIA_PRESENT)
+    if(CServiceBroker::GetMediaManager().GetDriveStatus(it->strPath) == DRIVE_CLOSED_MEDIA_PRESENT)
```
Comment (Rechi) [inline xbmc/platform/win32/storage/Win32StorageProvider.cpp]: space is missing between `if` and `(`

File: xbmc/platform/win32/storage/Win32StorageProvider.cpp [hunk 11]
```diff
@@ -362,16 +363,16 @@ bool CDetectDisc::DoWork()
   CLog::Log(LOGDEBUG, "%s: Optical media found in drive %s", __FUNCTION__, m_strPath.c_str());
   CMediaSource share;
   share.strPath = m_strPath;
-  share.strStatus = g_mediaManager.GetDiskLabel(share.strPath);
-  share.strDiskUniqueId = g_mediaManager.GetDiskUniqueId(share.strPath);
-  if(g_mediaManager.IsAudio(share.strPath))
+  share.strStatus = CServiceBroker::GetMediaManager().GetDiskLabel(share.strPath);
+  share.strDiskUniqueId = CServiceBroker::GetMediaManager().GetDiskUniqueId(share.strPath);
+  if(CServiceBroker::GetMediaManager().IsAudio(share.strPath))
```
Comment (Rechi) [inline xbmc/platform/win32/storage/Win32StorageProvider.cpp]: space is missing between `if` and `(`

File: xbmc/windows/GUIWindowFileManager.cpp [hunk 12]
```diff
@@ -686,7 +686,7 @@ bool CGUIWindowFileManager::HaveDiscOrConnection( std::string& strPath, int iDri
 {
   if ( iDriveType == CMediaSource::SOURCE_TYPE_DVD )
   {
-    if ( !g_mediaManager.IsDiscInDrive(strPath) )
+    if ( !CServiceBroker::GetMediaManager().IsDiscInDrive(strPath) )
```
Comment (Rechi) [inline xbmc/windows/GUIWindowFileManager.cpp]: no space after `(` and before `)`

**Review by AlwinEsch — APPROVED** [review]
Good to have this static `class CMediaManager g_mediaManager;` away.

File: xbmc/filesystem/CDDADirectory.cpp [hunk 13]
```diff
@@ -11,6 +11,7 @@
 #include "File.h"
 #include "FileItem.h"
 #include "music/MusicDatabase.h"
+#include "ServiceBroker.h"
```
Comment (Rechi) [inline xbmc/filesystem/CDDADirectory.cpp]: wrong include order

File: xbmc/filesystem/CDDAFile.cpp [hunk 14]
```diff
@@ -9,6 +9,7 @@
 #include "CDDAFile.h"
 
 #include "URL.h"
+#include "ServiceBroker.h"
```
Comment (Rechi) [inline xbmc/filesystem/CDDAFile.cpp]: wrong include order

File: xbmc/platform/win10/storage/Win10StorageProvider.cpp [hunk 15]
```diff
@@ -9,6 +9,7 @@
 
 #include "filesystem/SpecialProtocol.h"
 #include "guilib/LocalizeStrings.h"
+#include "ServiceBroker.h"
```
Comment (Rechi) [inline xbmc/platform/win10/storage/Win10StorageProvider.cpp]: wrong include order

File: xbmc/platform/win32/WIN32Util.cpp [hunk 16]
```diff
@@ -22,6 +22,7 @@
 #include "utils/StringUtils.h"
 #include "CompileInfo.h"
 #include "platform/win32/CharsetConverter.h"
+#include "ServiceBroker.h"
```
Comment (Rechi) [inline xbmc/platform/win32/WIN32Util.cpp]: wrong include order

File: xbmc/platform/win32/storage/Win32StorageProvider.cpp [hunk 17]
```diff
@@ -9,6 +9,7 @@
 
 #include "filesystem/SpecialProtocol.h"
 #include "guilib/LocalizeStrings.h"
+#include "ServiceBroker.h"
```
Comment (Rechi) [inline xbmc/platform/win32/storage/Win32StorageProvider.cpp]: wrong include order

File: xbmc/platform/win32/storage/Win32StorageProvider.cpp [hunk 18]
```diff
@@ -33,16 +34,24 @@ void CWin32StorageProvider::Initialize()
...
   GetDrivesByType(vShare, DVD_DRIVES);
   if(!vShare.empty())
-    g_mediaManager.SetHasOpticalDrive(true);
+    CServiceBroker::GetMediaManager().SetHasOpticalDrive(true);
   else
     CLog::Log(LOGDEBUG, "%s: No optical drive found.", __FUNCTION__);
 
 #ifdef HAS_DVD_DRIVE
   // Can be removed once the StorageHandler supports optical media
+<<<<<<< HEAD
```
Comment (Rechi) [inline xbmc/platform/win32/storage/Win32StorageProvider.cpp]: rebase error

File: xbmc/platform/win32/storage/Win32StorageProvider.cpp [hunk 19]
```diff
@@ -33,16 +34,24 @@ void CWin32StorageProvider::Initialize()
...
   for (const auto& it : vShare)
     if (g_mediaManager.GetDriveStatus(it.strPath) == DRIVE_CLOSED_MEDIA_PRESENT)
       CJobManager::GetInstance().AddJob(new CDetectDisc(it.strPath, false), NULL);
       // remove end
+=======
```
Comment (Rechi) [inline xbmc/platform/win32/storage/Win32StorageProvider.cpp]: rebase error

File: xbmc/platform/win32/storage/Win32StorageProvider.cpp [hunk 20]
```diff
@@ -33,16 +34,24 @@ void CWin32StorageProvider::Initialize()
...
+  VECSOURCES::const_iterator it;
+  for(it=vShare.begin();it!=vShare.end();++it)
+    if (CServiceBroker::GetMediaManager().GetDriveStatus(it->strPath) == DRIVE_CLOSED_MEDIA_PRESENT)
+      CJobManager::GetInstance().AddJob(new CDetectDisc(it->strPath, false), NULL);
+  // remove end
+>>>>>>> 16b804e359... remove g_mediaManager global
```
Comment (Rechi) [inline xbmc/platform/win32/storage/Win32StorageProvider.cpp]: rebase error

**Review by garbear — APPROVED** [review]

**Comment by lrusak** [comment]
jenkins build this please

**Comment by lrusak** [comment]
jenkins build this please

**Comment by lrusak** [comment]
merge?

**Review by garbear — APPROVED** [review]

**Comment by lrusak** [comment]
jenkins build this please

**Pull request merged**